source/js/payments.js

```diff
@@ -0,0 +1,116 @@
+                    styles: {},
+                    fields: {
+                        number: {
+                            selector: '#card-number',
+                        },
+                        cvv: {
+                            selector: '#cvv',
+                        },
+                        expirationDate: {
+                            selector: '#expiration-date',
+                        }
+                    }
+                };
+
+                hostedFields.create(options, function (hostedFieldsErr, hostedFieldsInstance) {
+                    if (hostedFieldsErr) {
+                        showErrorMessage(loadingErrorMsg);
+                        cardInputDiv.toggleAttribute('hidden', true);
+                        return;
+                    }
+
+                    submitButton.removeAttribute('disabled');
+
+                    hostedFieldsInstance.on('validityChange', function(event) {
+                        var field = event.fields[event.emittedBy];
+
+                        if (field.isValid || field.isPotentiallyValid) {
+                            clearFieldError(field.container);
+                        } else {
+                            showFieldError(field.container);
+                        }
+                    });
+
+                    submitButton.addEventListener('click', function(e){
+                        e.preventDefault();
+                        var state = hostedFieldsInstance.getState(),
+                            formValid = Object.keys(state.fields).every(function (key) {
+                                var isValid = state.fields[key].isValid;
+                                if(!isValid) {
+                                    showFieldError(state.fields[key].container);
+                                }
+                                return isValid;
+                            });
+
+                        if(formValid){
+                            clearErrorMessage('card');
+                            hostedFieldsInstance.tokenize(function(tokenizeErr, payload){
+                                if(tokenizeErr) {
+                                    showErrorMessage('There was an error processing your payment. Please try again.');
+                                    return;
+                                }
+
+                                nonceInput.value = payload.nonce;
+                                paymentModeInput.value = 'card';
+                                paymentForm.submit();
+                            });
+                        }
+                        else {
+                            showErrorMessage('Some of the fields below are invalid. Please correct the invalid fields and try again.');
+                        }
+                    });
+                });
+            }
+        );
+    }
+
+    initHostedFields();
+}
+
+document.addEventListener('DOMContentLoaded', function() {
+    setupBraintree();
+});
```

Comment:
Not quite sure how to review this :) Is there a javascript styleguide anywhere that we're supposed to be adhering to? It could be a good idea to have a front-end dev cast an eye over this.

Reply:
@chris-lawton would you be able to review this JS please? We have ESLint set up on the project but it looks like it has to be run separately - I will do that next week.

Reply:
Looks good @solarissmoke!

Reply:
Thanks! There are linting issues in this JS and in the styles, which I will make a separate PR for. Also need to add linting to travis.